src/compile/mark/point.ts

```diff
@@ -10,66 +14,77 @@ export namespace point {
   export function properties(model: UnitModel, fixedShape?: string) {
     // TODO Use Vega's marks properties interface
     let p: any = {};
+    const config = model.config();
 
-    // x
-    if (model.has(X)) {
-      p.x = {
-        scale: model.scaleName(X),
-        field: model.field(X, { binSuffix: '_mid' })
-      };
-    } else {
-      p.x = { value: model.config().scale.bandSize / 2 };
-    }
+    p.x = x(model.encoding().x, model.scaleName(X), config);
```

Comment:
For now, I'm trying not to pass `p` inside these methods and have side effect inside the methods.  
Let's see if this works well for other marks too.